### PR TITLE
Add API testing (And add 8 points for new opportunity)

### PR DESCRIPTION
**Pull request by sbetti02**
The end user changes from this PR is pretty minimal, it's just adding 8 points to alumni for posting a new opportunity.

The main bulk of this is adding in the infrastructure for writing tests that hit the API. It's definitely a work in progress, but this should allow people to get faster feedback on what they're building, and provide confidence that everything continues to work as new changes get added.

The main issue with the setup at the moment is that the test only works once you've taken out the required auth for the route for now. We definitely want to get this fixed sooner rather than later so we can run tests against the API easily as a test suite.

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -7,4 +7,17 @@
...
-From server directory, run `npm run test` to run all tests. Run `npm run test tests/<name_of_test>.test.js` to run a specific test suite.
\ No newline at end of file
+From server directory, run `npm run test` to run all tests. Run `npm run test tests/<name_of_test>.test.js` to run a specific test suite.
+
+**NOTE**
+
+For writing tests that hit the API endpoints, auth is not correctly set up at this time.
+When you want to check if tests for an endpoint are working, temporarily remove
+the passport middlewhere, such as
+```router.post('/opportunity/:id', async (req, res, next) => {```
```
Comment (reshadbinharun) [inline server/README.md:17]: Thanks for the note!

File: server/test/footyPoints.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+const mongoose = require("mongoose")
+const createServer = require("./testingTools")
+var alumniSchema = require('../models/alumniSchema');
+const request = require('supertest');
+
```
Comment (reshadbinharun) [inline server/test/footyPoints.test.js:5]: nit: extra line, I know, we need a linter setup

File: server/test/footyPoints.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * This runs before all tests in this particular file. I believe
+ * this will needed to be included in every test file to get them
+ * to work right.
+ * I'd bet there's a way to get this to run once per test suite
+ * instead of once per file, but I haven't looked into it yet.
+ */
+beforeAll(async () => {
+    process.env.NODE_ENV = 'test';
```
Comment (reshadbinharun) [inline server/test/footyPoints.test.js:17]: Can you add this variable in .env.tpl for /server?

File: server/test/footyPoints.test.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    mongoose.connect("mongodb://localhost:27017/test-test", { useNewUrlParser: true });
+    await request(app).get('/util/seed/');
```
Comment (reshadbinharun) [inline server/test/footyPoints.test.js:19]: Sometimes the /seed API fails due to duplicate key errors. When hitting this manually, it's sufficient to just clear all data and rehit the API.
This should be a easy fix on utilMongoose but we can leave a TODO and move on for now!

File: server/test/footyPoints.test.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ * Same thing as the beforeAll function, needs to be included on
+ * all test files where you're hitting API endponts.
+ * Just cleans up the tests after they've run.
+ * 
+ * The database might be better off getting reset after each test,
+ * rather than after all have run, to help each test run independently
+ * and be unaffected by surrounding tests (idempotent)
+ */
+afterAll(async () => {
+    const collections = Object.keys(mongoose.connection.collections);
```
Comment (reshadbinharun) [inline server/test/footyPoints.test.js:33]: love it!

File: server/test/testingTools.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+const express = require("express");
+var alumniRouter = require('../routes/alumni');
+var mongooseUtilRouter = require('../routes/utilMongoose');
+
+
```
Comment (reshadbinharun) [inline server/test/testingTools.js:5]: I see this is a deliberate stylistic thing you do. Me gusta

**Review by reshadbinharun — APPROVED** [review]
🚢 🚢

File: server/test/footyPoints.test.js (hunk 4 above)
Comment (sbetti02) [reply server/test/footyPoints.test.js:19]: Which key? The user ID?

File: server/test/footyPoints.test.js (hunk 2 above)
Comment (sbetti02) [reply server/test/footyPoints.test.js:5]: Saw your other comment about this about it being a stylistic choice, and yeah, that's kinda my take.. It's pretty common in Python to do it this way. That said, if in JS that's not the common way to do it, or for this project you want to keep it to one line, I'm happy to do so

**Review by craragon77 — APPROVED** [review]

**Pull request merged**